ghastly/region.py

```diff
@@ -0,0 +1,158 @@
+            more than one intake.
+        outtake : str
+            reg_id for the region that this one feeds into.  A region may only
+            have one outtake.
+        start : bool
+            True if this region is part of the "start" of a pebble's
+            path in the core - for example, if it is in the top of an HTGR -
+            False otherwise.
+        end : bool
+            True if this region is at the end of the pebble pathway through
+            the core - i.e., if it is the discharge region - False otherwise
+        '''
+        self.x_c = x_c
+        self.y_c = y_c
+        self.z_max = z_max
+        self.z_min = z_min
+        self.h = abs(z_min) + abs(z_max)
+        self.reg_id = reg_id
+        self.intake = intake
+        self.outtake = outtake
+        self.start = start
+        self.end = end
+
+
+class CylReg(Region):
+    '''
+    Class for a cylindrical region aligned with the z-axis.
+    '''
+
+    def __init__(self, r, *args, **kwargs):
+        '''
+        Initializes a single instance of a CylReg object.  All dimensions
+        should be in meters.
+
+        Parameters
+        ----------
+        r : float
+            Radius of the cylinder [m].
+        '''
+        super().__init__(*args, **kwargs)
+        self.r = r
+        self.volume = np.pi * (r**2) * self.h
+
+
+class AnnularReg(Region):
+    '''
+    Class for a sector of an annular region, to be used with CylCore objects.
+    '''
+
+    def __init__(self, r_outer, r_inner, theta_min=0, theta_max=2 * np.pi,
+                 *args, **kwargs):
+        '''
+        Initializes a single instance of an AnnularReg object.  All distances
+        should be in meters, all angles in radians.
+
+        Parameters
+        ----------
+        r_outer : float
+            Outer radius of the annulus [m].
+        r_inner : float
+            Inner radius of the annulus [m].
+        theta_min : float
+            Smaller angle that defines the sector of the annular region.
+            Default is 0 radians.
+        theta_max : float
+            Larger angle that defines the sector of the annular region.
+            Default is 2pi radians.
+        '''
+        super().__init__(*args, **kwargs)
+        self.r_outer = r_outer
+        self.r_inner = r_inner
+        self.theta_min = theta_min
+        self.theta_max = theta_max
+        self.volume = ((np.pi * (r_outer**2 - r_inner**2) * self.h)
+                       * ((theta_max - theta_min) / (2 * np.pi)))
+
+
+class ConeReg(Region):
+    '''
+    Class for a region in the shape of a truncated right cone.
+    '''
+
+    def __init__(self, r_upper, r_lower, *args, **kwargs):
+        '''
+        Initializes a ConeReg object.  All dimensions should be in meters.
+
+        Parameters
+        ----------
+        r_upper : float
+            Radius at the top of the cone [m].
+        r_lower : float
+            Radius at the bottom of the cone [m].
```

Comment:
Consistency with ConeCore (Does not necessarily correspond to the largest radius.)